### PR TITLE
GUAC-1389: Add screen sharing support to RDP.

### PR DESCRIPTION
**Pull request by mike-jumper**

### CODE REVIEW

File: src/protocols/rdp/rdp.c [hunk 1]
```diff
@@ -0,0 +1,931 @@
...
+                &write_count)) {
+        guac_client_abort(client, GUAC_PROTOCOL_STATUS_SERVER_ERROR, "Unable to read RDP channel file descriptors.");
+        return -1;
+    }
+
+    /* Construct read fd_set */
+    max_fd = 0;
+    FD_ZERO(&rfds);
+    for (index = 0; index < read_count; index++) {
+        fd = (int)(long) (read_fds[index]);
```
Comment (jmuehlner) [inline src/protocols/rdp/rdp.c]: Why (int)(long)?

File: src/protocols/rdp/rdp.c [hunk 2]
```diff
@@ -0,0 +1,931 @@
...
+
+#ifdef HAVE_FREERDP_DISPLAY_UPDATE_SUPPORT
+        /* Update remote display size */
+        pthread_mutex_lock(&(rdp_client->rdp_lock));
+        guac_rdp_disp_update_size(rdp_client->disp, rdp_inst->context);
+        pthread_mutex_unlock(&(rdp_client->rdp_lock));
+#endif
+
+        /* Wait for data and construct a reasonable frame */
+        int wait_result = rdp_guac_client_wait_for_messages(client, 250000);
```
Comment (jmuehlner) [inline src/protocols/rdp/rdp.c]: Might be nice to document this magic number a bit too...

File: src/protocols/rdp/rdp.c (hunk 1 above)
Comment (mike-jumper) [inline src/protocols/rdp/rdp.c]: Well...

`read_fds` and `write_fds` in this breed of functions from FreeRDP are both `void**`. Typecasting those directly to `int` can't work:

```
test.c:9:14: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
```

Typecasting from `void**` to `long` to `int` will work, however.

File: src/protocols/rdp/rdp.c (hunk 2 above)
Comment (mike-jumper) [inline src/protocols/rdp/rdp.c]: :tophat:

**Pull request merged**